napalm/nxos_ssh/nxos_ssh.py

```diff
@@ -791,6 +791,7 @@ def get_facts(self):
                 break
         if hostname.count(".") >= 2:
             fqdn = hostname
+            hostname = hostname.split(".")[0]
```

Comment:
Hostname itself can contain dots, this code will then fail splitting domain name from hostname. For example fqdn: "my.nexus.spam.com" will give hostname my, while expected result should be "my.nexus". More intelligent parsing should be applied involving getting domain name from the box.

Reply:
Good point. Will fix.